### PR TITLE
[DEV-2568] Dubai transactions improvements

### PR DESCRIPTION
**Pull request by lapatric**

### CODE REVIEW

**Review by TaprootFreak — APPROVED** [review]

File: src/screens/sell-info.screen.tsx [hunk 1]
```diff
@@ -183,6 +229,18 @@ export function SellInfoScreen(): JSX.Element {
...
+  function getPaymentInfoString(paymentInfo: Sell, selectedBankAccount: BankAccount): string {
+    return (
+      paymentInfo &&
+      selectedBankAccount &&
+      translate('screens/payment', 'Please send the specified amount to the address below.', {
+        chain: toString(paymentInfo.asset.blockchain),
+        currency: paymentInfo.currency.name,
+        iban: Utils.formatIban(selectedBankAccount.iban) ?? '',
+      })
+    );
```
Comment (davidleomay) [inline src/screens/sell-info.screen.tsx:241]: Looks like the translation parameters are not needed?

File: src/screens/sell-info.screen.tsx [hunk 2]
```diff
@@ -169,6 +205,16 @@ export function SellInfoScreen(): JSX.Element {
...
 
+  async function checkTransaction() {
+    if (!paymentInfo || sellTxId) return;
+    getTransactionByRequestId(paymentInfo.id.toString())
+      .then((tx) => {
+        setSellTxId(tx.id.toString());
+        setShowsCompletion(true);
+      })
+      .catch(() => setTimeout(checkTransaction, 5000));
+  }
```
Comment (davidleomay) [inline src/screens/sell-info.screen.tsx:216]: We should probably cancel this "interval" when the payment info changes?

File: src/screens/sell-info.screen.tsx [hunk 3]
```diff
@@ -169,6 +205,16 @@ export function SellInfoScreen(): JSX.Element {
     return sell;
   }
 
+  async function checkTransaction() {
+    if (!paymentInfo || sellTxId) return;
+    getTransactionByRequestId(paymentInfo.id.toString())
+      .then((tx) => {
+        setSellTxId(tx.id.toString());
```
Comment (davidleomay) [inline src/screens/sell-info.screen.tsx:212]: `setSellTxId(tx.inputTxId);`

File: src/screens/sell-info.screen.tsx [hunk 4]
```diff
@@ -109,6 +122,29 @@ export function SellInfoScreen(): JSX.Element {
     }
   }, [bankAccountParam, getAccount, bankAccounts, countries]);
 
+  useEffect(() => {
+    if (!paymentInfo) return;
+    const priceTimestamp = new Date(paymentInfo.timestamp).getTime();
+    const interval = setInterval(() => {
+      const now = Date.now();
+      const expiration = priceTimestamp + Math.ceil((now - priceTimestamp) / 900_000) * 900_000;
```
Comment (davidleomay) [inline src/screens/sell-info.screen.tsx:130]: Isn't it just
```ts
const priceTimestamp = new Date(paymentInfo.timestamp);
const expiration = priceTimestamp.setMinutes(priceTimestamp.getMinutes() + 15);
```
?

File: src/screens/sell-info.screen.tsx [hunk 5]
```diff
@@ -218,75 +278,85 @@ export function SellInfoScreen(): JSX.Element {
...
+                    )})`}
+                  >
+                    {paymentInfo.beneficiary.iban}
+                    <CopyButton onCopy={() => copy(bankAccount.iban)} />
+                  </StyledDataTableRow>
+                  {paymentInfo.beneficiary.name && (
+                    <StyledDataTableRow label={translate('screens/payment', 'Beneficiary Bank Account Name')}>
+                      {paymentInfo.beneficiary.name}
+                      <CopyButton onCopy={() => copy(bankAccount.iban)} />
+                    </StyledDataTableRow>
```
Comment (davidleomay) [inline src/screens/sell-info.screen.tsx:308]: Update copy() params

**Review by davidleomay — APPROVED** [review]

**Pull request merged**